### DISCUSSION TITLE
Linak

### DISCUSSION THREAD

**Discussion by alanmiller** [tjhorner/wifi-desk-controller#20] [Reverse Engineering Help]
Hi I just got my UpsyDesky but it doesn't seem to be detecting the desk controller.
I'd like to help get this desk working if possible, but I'm not sure of the next steps to reverse engineer.

I don't use HomeAssistant, I was planning on just writing some Python to use the HTTP API.
For now, all I get is:  `{"id":"sensor-desk_height","value":null,"state":"NA"}` when calling `http://172.16.1.13/sensor/desk_height`

I've attached 2 pics
1. the nameplate from my controller and 
2. the keypad connected to the left port on the upsydesky and the right port connected to the controller

It seems the keypad only uses 4 pins the 1st, and last 3 pins (from left to right)
When connected in this manner the keypad still works (up, down, 3 presets).
I also flashed the firmware.bin from the webui (OTA update said it was successful).

![linak-nameplate](https://user-images.githubusercontent.com/1309167/193698228-961cdbf0-f2d0-4d16-931d-a31aaf0e7d0a.png)
![linak-upsy-desky](https://user-images.githubusercontent.com/1309167/193698257-9ee0b421-1350-489a-8ba3-730fbaead06c.png)

**Comment by ImCiryus** [comment]
Hey! Have you ever gotten it to work? Running into a similar problem here but I'm still looking into it

**Reply by FqfxLSNiKo** [reply]
Me too. Did you happen to get it working?

**Reply by alanmiller** [reply]
Nope. I bought a DPG1C32 instead.

**Comment by FqfxLSNiKo** [comment]
@alanmiller Did you happen to get it working?

**Reply by alanmiller** [reply]
Nope. I bought a DPG1C32 instead.

**Reply by FqfxLSNiKo** [reply]
Ok, thanks. Did you manage to integrate it into Home Assistant? Could you explain how?
